lib/modules/manager/sveltos/extract.ts

```diff
@@ -0,0 +1,102 @@
+import is from '@sindresorhus/is';
+import { logger } from '../../../logger';
+import { coerceArray } from '../../../util/array';
+import { trimTrailingSlash } from '../../../util/url';
+import { parseYaml } from '../../../util/yaml';
+import { DockerDatasource } from '../../datasource/docker';
+import { HelmDatasource } from '../../datasource/helm';
+import { getDep } from '../dockerfile/extract';
+import { isOCIRegistry, removeOCIPrefix } from '../helmv3/oci';
+import type {
+  ExtractConfig,
+  PackageDependency,
+  PackageFileContent,
+} from '../types';
+import { ProfileDefinition, type SveltosHelmSource } from './schema';
+import { removeRepositoryName } from './util';
+
+export function extractPackageFile(
+  content: string,
+  packageFile: string,
+  config?: ExtractConfig,
+): PackageFileContent | null {
+  let definitions: ProfileDefinition[];
+  try {
+    definitions = parseYaml(content, {
+      customSchema: ProfileDefinition,
+    });
+  } catch (err) {
+    logger.debug({ err, packageFile }, 'Failed to parse Sveltos definition.');
+    return null;
+  }
+
+  const deps = definitions.flatMap((definition) =>
+    extractDefinition(definition, config),
+  );
+
+  return deps.length ? { deps } : null;
+}
+
+export function extractDefinition(
+  definition: ProfileDefinition,
+  config?: ExtractConfig,
+): PackageDependency[] {
+  const result = ProfileDefinition.safeParse(definition);
```

Comment:
why parsing again? makes no sense as it's already validated in caller. remove and inline this function.